Resources/Private/JavaScript/SlubMediaPlayer/components/equalizer/EqualizerPlugin.js

```diff
@@ -66,8 +67,45 @@ export default class EqualizerPlugin extends DlfMediaPlugin {
    * @param {DlfMediaPlayer} player
    */
   async attachToPlayer(player) {
-    if (window.location.protocol !== "https:") {
-      console.error("Warning: The equalizer will probably fail without HTTPS");
+    // If the equalizer element is inside a hidden panel (e.g. parent has
+    // `hidden`), defer initialization until it becomes visible. This avoids
+    // initializing/resizing the canvas while the element has zero width.
+    const hostPanel = this.closest('[data-panel]');
+    const isHidden = hostPanel && hostPanel.hasAttribute('hidden');
+
+    if (isHidden && hostPanel instanceof HTMLElement) {
+      // Defer init and observe the host panel for attribute changes
+      const obs = new MutationObserver((records, observer) => {
+        for (const r of records) {
+          if (r.type === 'attributes' && r.attributeName === 'hidden') {
+            if (!hostPanel.hasAttribute('hidden')) {
+              observer.disconnect();
+              // Fire-and-forget initialization
+              this.initForPlayer(player).catch((e) => console.error('Equalizer init failed:', e));
```

Comment:
Is this working as intended?
I am not very familiar with this, but Codacy warns about a syntax error and MDN suggests something like this:
```suggestion
              this.initForPlayer(player).catch((e) => { console.error('Equalizer init failed:', e) });
```